### ISSUE TITLE
Conditionals ignored when block is inside an include

### ISSUE THREAD

**Issue by dissnn** [ansible/ansible#16952]
##### ISSUE TYPE
- Bug Report
##### ANSIBLE VERSION

```
2.1.1.0-1ppa~trusty
```
##### OS / ENVIRONMENT

Ubuntu 14.04
##### SUMMARY

Conditional check `when` don't working for `block` in included task that included after another `include` that used  `with_items`. Look at `STEPS TO REPRODUCE` below.
##### STEPS TO REPRODUCE
- **1.yml**

```
- hosts: localhost
  tasks:
    - include: 3.yml variable={{ item }}
      with_items:
        - test_var

    - include: 2.yml

```
- **2.yml**

```

---

- block:
  - debug: msg='I run when test_var is True ( test_var = {{ test_var }} )'
  when: test_var
```
- **3.yml**

```

---

- debug: msg='Never mind...'
```

Let's run `ansible-playbook`

```
ansible-playbook  1.yml -e '{ test_var: False }'
```
##### EXPECTED RESULTS

Task skipped by "Conditional check failed"
##### ACTUAL RESULTS

Task don't skipped

```
$ ansible-playbook 1.yml  -e '{test_var: True}'
Using /etc/ansible/ansible.cfg as config file
 [WARNING]: provided hosts list is empty, only localhost is available


PLAYBOOK: 1.yml ****************************************************************
1 plays in 1.yml

PLAY [localhost] ***************************************************************

TASK [setup] *******************************************************************
ok: [localhost]

TASK [include] *****************************************************************
included: /home/dissnn/3.yml for localhost

TASK [debug] *******************************************************************
ok: [localhost] => {
    "msg": "Never mind..."
}

TASK [debug] *******************************************************************
ok: [localhost] => {
    "msg": "I run when test_var is True ( test_var = False )"
}

PLAY RECAP *********************************************************************
localhost                  : ok=4    changed=0    unreachable=0    failed=0  
```

**Comment by xenithorb** [comment]
This is also happening to me with one of my roles where I have a pretty basic `main.yml` and am subsequently including tasks from that. The `main.yml` is pretty generic and easy to read so I'm not going to generalize it: 

``` yaml
---
# tasks file for graymonarch.freeswitch

- name: Include distribution specific variables
  include_vars: "{{ item }}"
  with_first_found:
    - "{{ ansible_distribution }}.yml"
    - "{{ ansible_os_family }}.yml"

# Currently with_first_found: does not discriminate folders,
# so we can't have two RedHat.yml's, one must be uniquely named
# https://github.com/ansible/ansible/issues/14857
- name: Include tasks from distributions specific tasks
  include: "{{ item }}"
  with_first_found:
    - "t_{{ ansible_distribution }}.yml"
    - "t_{{ ansible_os_family }}.yml"

- name: Include install tasks
  include: install.yml

- name: ODBC + DSN Setup for MySQL/MariaDB and Freeswitch
  include: odbc_setup.yml
  when: gm_fs_odbc_do_setup

- name: Include service tasks
  include: service.yml
```

The error occurs in `odbc_setup.yml` where there is a block with a `when: condition` and it just runs the block anyway, even if I set it to something that _must_ be false. (like 0==1 or "false")

**Comment by xenithorb** [comment]
@bcoca I don't think this is labeled correctly as a "feature_idea"

**Comment by riccardomurri** [comment]
It looks like this affects Ansible 2.1.1.0 but _not_ 2.1.0.0.

**Comment by bcoca** [comment]
So the simple case works:

```
- hosts: localhost
  gather_facts: no
  tasks:
    - include: 2.yml
```

but for some strange reason it gets 'lost' when another include is put in, at least in 2.1

**Comment by bcoca** [comment]
tested against devel, so this has already been fixed (probably a side effect of proper scoping) and will be released in 2.2
